lib/streamlit/runtime/pages_manager.py

```diff
@@ -51,7 +51,56 @@ def _on_pages_changed(_path: str) -> None:
                 glob_pattern="*.py",
                 allow_nonexistent=True,
             )
-            PagesManagerV1.is_watching_pages_dir = True
+            PagesStrategyV1.is_watching_pages_dir = True
+
+    def __init__(self, pages_manager: PagesManager, **kwargs):
+        self.pages_manager = pages_manager
+
+        if kwargs.get("setup_watcher", True):
```

Comment:
nit: this reads a bit weirdly, could we extract `setup_watcher` with a default value of `True` in the method signature before capturing the rest of the `kwargs`? or is there some reason why writing things this way is necessary (if so, it'd be good to add an explanatory comment)

Reply:
Yea, I think my original intention was to have some custom args associated with different strategies, but I realized this is too much needed. I'll stick with the default argument.